airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -431,13 +448,17 @@ def _start_dataflow(
         variables: Dict,
         name: str,
         command_prefix: List[str],
-        label_formatter: Callable[[Dict], List[str]],
         project_id: str,
         multiple_jobs: bool = False,
         on_new_job_id_callback: Optional[Callable[[str], None]] = None,
         location: str = DEFAULT_DATAFLOW_LOCATION
     ) -> None:
-        cmd = command_prefix + self._build_cmd(variables, label_formatter, project_id)
+        cmd = command_prefix + [
+            "--runner=DataflowRunner",
```

Comment:
This makes me think (larger scope than just SQL operator) of should we have Beam Operators that support other runners?

For example some users it does not make sense Dataflow for smaller/shorter batch jobs say (because you have the overhead of waiting for workers to come up) For a job < 30 mins worker spin up time can be 10% performance hit. But they may still want to use Apache Beam (on say spark runner) that submits to non-ephemeral cluster (dataproc, EMR, spark on k8s, on prem infra, etc). 

Would this be easy enough to achieve on Dataproc / EMR / Spark Operators ?

Reply:
It will be quite huge task to write Apache Beam Hooks and Operators but worth to keep it in mind to do this in future.

Reply:
In this Q, we want to start working on operators for Apache Beam.